script.js

```diff
@@ -58,62 +59,87 @@
     return uniq([...pluralized, ...possibleAliases])
   }
 
-  const enrichResources = (resources) =>
+   enrichResources = (resources) => (
     resources.map((a) =>
       ({ ...a, extraTags: addTagAliases(a.tags) }))
+  )
 
   const buildResourceItem = ({ href, title, tags, description }) => {
     const link = `<a href="${href}" rel="noopener noreferrer nofollower" target="_blank">${title}</a/>`
     const tagList = `<small>tags: ${tags.join(' ')}</small>`
     return `<li>${[link, description, tagList].filter(Boolean).join('<br>')}</li>`
   }
 
+  const fetchy = (resources) => {
+    // enrich resource with plurals and tag aliases from functions above
+    const enrichedResources = enrichResources(resList)
+
+    const fuse = new window.Fuse(enrichResources(resList), fuseOptions)
+
+    // set up filtering code
+    const addListFilter = () => {
+      const ul = $(document.body).find('ul')
+
+      // actual filtering function
+      const filterRows = (e) => {
+        //console.log("I wanna show all of it!")
+  
+        var completeList = itemsList.concat(items)
+        //console.log(completeList)
+
+        $('ul').html(completeList)
+        addListFilter()
+
+        const query = document.getElementById("query").value.trim()
+        //const query = e.target.value.trim()
+        // show all items if no query
+        if (query === '') {
+          ul.find('li').show()
+        } else {
+          
+          const fuseResults = fuse.search(query)
+            .map((a) => a.item)
+            .map((a) => a.href)
+          const allLis = ul.find('li')
+          // assume nothing found, then just show found rows
+          // this should be refactored
+          ul.find('li').hide()
+          allLis.filter((i, element) => {
+            const href = $(element).find('a').attr('href')
+            return fuseResults.find((l) => l === href)
+          }).show()
+        }
+      }
+
+      $('#filter').on('click', filterRows)
+    }
+
+    // build the elements from the resources, and flip it since new
+    // links are added to the end of the yaml file
+    const items = enrichedResources.map(buildResourceItem).reverse()
+
+    const itemsList = items.splice(0,15)
```

Comment:
```suggestion
    const itemsList = items.slice(0, 15)
```

`splice` mutates the original array, `slice` does not. These splices progressively delete more of the original array. Need to take into account the current index and use a `slice`.